src/graph/UpdateEdgeExecutor.cpp

```diff
@@ -0,0 +1,258 @@
+        auto propName = item->field();
+        updateItem.prop = *propName;
+        updateItem.value = Expression::encode(item->value());
+        updateItems_.emplace_back(std::move(updateItem));
+    }
+    if (updateItems_.empty()) {
+        status = Status::Error("There must be some correct update items.");
+    }
+    return status;
+}
+
+
+Status UpdateEdgeExecutor::prepareWhen() {
+    auto *clause = sentence_->whenClause();
+    if (clause != nullptr) {
+        filter_ = clause->filter();
+    }
+    return Status::OK();
+}
+
+
+Status UpdateEdgeExecutor::prepareYield() {
+    auto *clause = sentence_->yieldClause();
+    if (clause != nullptr) {
+        yields_ = clause->columns();
+    }
+    return Status::OK();
+}
+
+
+std::vector<std::string> UpdateEdgeExecutor::getReturnColumns() {
+    std::vector<std::string> returnColumns;
+    for (auto *col : yields_) {
+        auto column = Expression::encode(col->expr());
+        returnColumns.emplace_back(std::move(column));
+    }
+    return returnColumns;
+}
+
+
+void UpdateEdgeExecutor::finishExecution(storage::cpp2::UpdateResponse &&rpcResp) {
+    resp_ = std::make_unique<cpp2::ExecutionResponse>();
+    std::vector<std::string> columnNames;
+    columnNames.reserve(yields_.size());
+    for (auto *col : yields_) {
+        if (col->alias() == nullptr) {
+            columnNames.emplace_back(col->expr()->toString());
+        } else {
+            columnNames.emplace_back(*col->alias());
+        }
+    }
+    resp_->set_column_names(std::move(columnNames));
+
+    std::vector<cpp2::RowValue> rows;
+    if (rpcResp.__isset.schema && rpcResp.__isset.data) {
+        std::shared_ptr<ResultSchemaProvider> schema;
+        if (rpcResp.get_schema() != nullptr) {
+            schema = std::make_shared<ResultSchemaProvider>(rpcResp.schema);
+        }
+        auto reader = RowReader::getRowReader(rpcResp.data, schema);
+        std::vector<cpp2::ColumnValue> row(yields_.size());
+        for (auto index = 0UL; index < yields_.size(); index++) {
+            auto res = RowReader::getPropByIndex(reader.get(), index);
+            if (ok(res)) {
+                auto column = value(std::move(res));
+                switch (column.which()) {
+                    case VAR_INT64:
+                        row[index].set_integer(boost::get<int64_t>(column));
+                        break;
+                    case VAR_DOUBLE:
+                        row[index].set_double_precision(boost::get<double>(column));
+                        break;
+                    case VAR_BOOL:
+                        row[index].set_bool_val(boost::get<bool>(column));
+                        break;
+                    case VAR_STR:
+                        row[index].set_str(boost::get<std::string>(column));
+                        break;
+                    default:
+                        LOG(FATAL) << "Unknown VariantType: " << column.which();
+                }
+            }
```

Comment:
if ok(res) is false, maybe you need insert a default value, else the row is not match the yields_ fields.

Reply:
yep, if ok(res) is false, it will do `onError_`.